presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -7889,10 +7890,16 @@ public void testQuantifiedComparisonCornerCases(String query)
         );
         List<QueryTemplate.Parameter> quantifier = new QueryTemplate.Parameter("quantifier").of("ALL", "ANY");
         List<QueryTemplate.Parameter> lhs = new QueryTemplate.Parameter("lhs").of("1", "NULL");
-        List<QueryTemplate.Parameter> operator = new QueryTemplate.Parameter("operator").of("<", ">", "<=", ">=");
+        List<QueryTemplate.Parameter> operator = new QueryTemplate.Parameter("operator").of("=", "!=", "<", ">", "<=", ">=");
         QueryTemplate queryTemplate = new QueryTemplate("SELECT %lhs% %operator% %quantifier% (%subquery%)");
+        ImmutableList<String> excludedInPredicateQueries = ImmutableList.of(
+                "SELECT NULL != ALL (SELECT * FROM (SELECT 1 WHERE false) as empty_table)",
```

Comment:
This seems brittle. Maybe we should introduce a mechanism for QueryTemplate to skip certain combinations of parameters and express these exclusions using that. So, something like:

```java
queryTemplate.with(
    parameter("subquery").of(...),
    parameter("quantifier").of(...),
    ...)
   .exclude(
      parameter("value").of("NULL"),
      parameter("operator").of("="),
      parameter("quantifier").of("ANY"),
      parameter("subquery").of("SELECT 1 WHERE false")),
   .exclude(
      parameter("value").of("NULL"),
      parameter("operator").of("!="),
      parameter("quantifier").of("ALL"),
      parameter("subquery").of("SELECT 1 WHERE false"));
```

Just a thought...

No need to do it now, though. I want to get this fix to the correctness issue in ASAP.